src/XrdPosix/XrdPosixLinkage.hh

```diff
@@ -296,26 +374,44 @@
 #define Retv_Stat64 int
 #define Args_Stat64 int, const char *, struct stat64 *
 #else
+#ifdef __APPLE__
+#define Symb_Stat64 UNIX_PFX "stat"
+#define Retv_Stat64 int
+#define Args_Stat64 const char *, struct stat64 *
+#else
 #define Symb_Stat64 UNIX_PFX "stat64"
 #define Retv_Stat64 int
 #define Args_Stat64 const char *, struct stat64 *
 #endif
+#endif
 
 #define Symb_Statfs    UNIX_PFX "statfs"
 #define Retv_Statfs    int
 #define Args_Statfs    const char *, struct statfs *
 
+#ifdef __APPLE__
+#define Symb_Statfs64  UNIX_PFX "statfs"
+#define Retv_Statfs64  int
+#define Args_Statfs64  const char *, struct statfs64 *
```

Comment:
Could we try following Apple's recommended approach? From the man page for `statfs`/`statfs64`:
```man
The statfs64 and fstatfs64 routines are equivalent to their corresponding non-64-suffixed
routine, when 64-bit inodes are in effect.  They were added before there was support for
the symbol variants, and so are now deprecated.  Instead of using these, set the 
_DARWIN_USE_64_BIT_INODE macro before including header files to force 64-bit inode support.
```